src/MongoDB/Monitoring/ServerOpeningEvent.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <Zend/zend_interfaces.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "phongo_compat.h"
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_serveropeningevent_ce;
+
+/* {{{ proto string ServerOpeningEvent::getHost()
+   Returns this event's host */
+static PHP_METHOD(ServerOpeningEvent, getHost)
+{
+	php_phongo_serveropeningevent_t* intern = Z_SERVEROPENINGEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_STRING((intern->host)->host);
```

Comment:
Note that the host _string_ is not the only vital information in [`mongoc_host_list_t`](http://mongoc.org/libmongoc/1.17.5/mongoc_host_list_t.html). The port is equally important, which is why we also have a `Server::getPort()` method. I think you'll need to add `getPort()` on this class as well.

I realize `mongoc_host_list_t` also has a `family` field; however, we don't currently make use of that. The libmongoc docs also don't go into much detail about that field (reported in [CDRIVER-4133](https://jira.mongodb.org/browse/CDRIVER-4133)), but I believe it's used to indicate the socket type (e.g. IPv4, IPv6, Unix domain socket). Nothing to worry about in the context of this PR, though. I've also opened [PHPC-1955](https://jira.mongodb.org/browse/PHPC-1955) to track this.